[JIT] Expose magic methods on script::Object

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38167 [JIT] Expose magic methods on script::Object**

Differential Revision: [D21486709](https://our.internmc.facebook.com/intern/diff/D21486709)

Commit:
[JIT] Expose magic methods on script::Object

[ghstack-poisoned]

Commit:
Update on "[JIT] Expose magic methods on script::Object"


Differential Revision: [D21486709](https://our.internmc.facebook.com/intern/diff/D21486709)

[ghstack-poisoned]

Commit:
Update on "[JIT] Expose magic methods on script::Object"


Differential Revision: [D21486709](https://our.internmc.facebook.com/intern/diff/D21486709)

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/python/script_init.cpp:
```diff
@@ -684,7 +733,16 @@ void initJitScriptBindings(PyObject* module) {
   py::class_<c10::intrusive_ptr<CustomClassHolder>>(m, "Capsule");
 
-  py::class_<Object>(m, "ScriptObject")
+#define DEF_MAGIC(name) \
```
There is no need for a macro or code gen here, a simple loop over the names, using a captured string would result in lower compile time and be easier to follow.

Commit:
Update on "[JIT] Expose magic methods on script::Object"


Differential Revision: [D21486709](https://our.internmc.facebook.com/intern/diff/D21486709)

[ghstack-poisoned]
